src/libstore/daemon.cc

```diff
@@ -222,7 +222,15 @@ struct ClientSettings
                     else if (!hasSuffix(s, "/") && trusted.count(s + "/"))
                         subs.push_back(s + "/");
                     else
-                        warn("ignoring untrusted substituter '%s'", s);
+                        warn("warning: trying to use an untrusted substituter '%s', but you are not a trusted user.\n\n"
+                             "/!\\ A trusted but malicious actor could hack your system through this substituter.\n\n"
```

Comment:
This being a warning has also been frustrating, often it is critical to use the provided substituter.

Reply:
"warning:" is superfluous because `Logger::warn()` already adds that (see `logging.cc`).

Reply:
I've drop the extra warning word.
As for the critical level, I'm not sure it's fitting because that doesn't prevent Nix to continue (and maybe compile everything). However, the behavior could be changed to make nix die when in this situation, instead of only displaying a warning.